### PR TITLE
[ACR] Add a validator for argument 'registry_name'

### LINKED ISSUE #12631: Omit login server endpoint for `--registry` on `az acr`
**Is your feature request related to a problem? Please describe.**
Many customers are frustrated in the experience of running the `az acr` command like below:
```
$ az acr run -f tasks.yaml . --registry myregistry.azurecr.io
The resource with name 'myregistry.azurecr.io' and type 'Microsoft.ContainerRegistry/registries' could not be found in subscription '<my subscription>'.
```
The correct input for `--registry` is `myregistry` but many customers mistype it with the login server `myregistry.azurecr.io`. When the customers read the error message, they are confused.

**Describe the solution you'd like**
To less confuse the customers, the `az acr` command sets should omit the `.azurecr.io` login server suffix for the customers if present.

**Describe alternatives you've considered**
If the above solution cannot be fulfilled, the `az acr` should at least prompt a friendly note to ask the customers omit the suffix by themselves instead of the confusing error message.

**Additional context**
Note that the domain prefix is available in
```
$ az cloud show --query "suffixes.acrLoginServerEndpoint"
".azurecr.io"
```

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolve #12631
Add a validator for argument 'registry_name' to omit login server endpoint suffix.
The validator applies to all `az acr` commands, except `az acr create` and `az acr check-name`. 

**Behavior**
(specify registry name with --registry)
image

(specify login server with --registry)
image

(specify login server with --name)
image

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py [hunk 1]
```diff
@@ -85,3 +88,14 @@ def validate_retention_days(namespace):
     days = namespace.days
     if days and (days < 0 or days > 365):
         raise CLIError("Invalid value for days: should be from 0 to 365")
+
+
+def validate_registry_name(cmd, namespace):
+    """Omit login server endpoint suffix."""
+    registry = namespace.registry_name
+    if registry:
+        suffix = cmd.cli_ctx.cloud.suffixes.acr_login_server_endpoint
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acr/_validators.py]: seems not all clouds define the `acr_login_server_endpoint`(AzureGermanCloud), it's better to have an empty check here.

Comment (Wwwsylvia) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: Thanks!

Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: AzureGermanCloud is defined here, https://github.com/Azure/azure-cli/blob/31386c8391217f2a5fcdf0016e80082c81a6ae30/src/azure-cli-core/azure/cli/core/cloud.py#L312. You can also find the other default clouds definition in the same file.

Comment (Wwwsylvia) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: It's very helpful, thanks a lot!

**Comment by yungezz** [comment]
HI @Wwwsylvia just fyi, next azure CLI release CC date is 3/25 UTC +8, pls help to address comments and let us know when the PR ready for reviewing/merging.

**Comment by Wwwsylvia** [comment]
> HI @Wwwsylvia just fyi, next azure CLI release CC date is 3/25 UTC +8, pls help to address comments and let us know when the PR ready for reviewing/merging.

Noted, thanks!

**Pull request merged**